client-spark/common/src/main/java/org/apache/spark/shuffle/writer/TaskAttemptAssignment.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.writer;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.spark.shuffle.handle.MutableShuffleHandleInfo;
+import org.apache.spark.shuffle.handle.ShuffleHandleInfo;
+
+import org.apache.uniffle.common.ShuffleServerInfo;
+import org.apache.uniffle.common.exception.RssException;
+
+/** This class is to get the partition assignment for ShuffleWriter. */
+public class TaskAttemptAssignment {
+  private Map<Integer, List<ShuffleServerInfo>> assignment;
+  private boolean mutable = false;
+
+  public TaskAttemptAssignment(long taskAttemptId, ShuffleHandleInfo shuffleHandleInfo) {
+    this.assignment = shuffleHandleInfo.getAvailablePartitionServersForWriter();
+    if (shuffleHandleInfo instanceof MutableShuffleHandleInfo) {
+      this.mutable = true;
+    }
+  }
+
+  /**
+   * Retrieving the partition's current available shuffleServers.
+   *
+   * @param partitionId
+   * @return
+   */
+  public List<ShuffleServerInfo> retrieve(int partitionId) {
+    return assignment.get(partitionId);
+  }
+
+  public void update(ShuffleHandleInfo handle) {
+    if (!mutable) {
```

Comment:
This is still confusing, `mutable` is derived from the handle given to the constructor, but `getAvailablePartitionServersForWriter` is called on the handle given to this method. There is no need for `mutable`, as this works perfectly fine with `ShuffleHandleInfo`.

Reply:
Yes. This could be removed.

Reply:
Done